dlock/dlock_test.go

```diff
@@ -0,0 +1,48 @@
+package dlock
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/lieut-data/mattermost-plugin-api/dlock/dlocktest"
+	"github.com/stretchr/testify/require"
+)
+
+// TODO(ilgooz): test all branches including related ones to Store errors and ExpireInSeconds.
+// TODO(ilgooz): can move tests from sync/mutex_test.go.
+
+func TestLock(t *testing.T) {
+	dl := New("a", dlocktest.New())
+	var wg sync.WaitGroup
+	wg.Add(3)
+	for i := 0; i < 3; i++ {
+		go func() {
+			defer wg.Done()
+			for i := 0; i < 10; i++ {
+				dl.Lock(context.Background())
```

Comment:
Maybe alter the test to also check the rough order in which the locks are acquired.